topostats/tracing/dnatracing.py

```diff
@@ -723,11 +787,16 @@ def trace_image(
     # Check both arrays are the same shape
     assert image.shape == grains_mask.shape
 
-    cropped_images, cropped_masks = prep_arrays(image, grains_mask, pad_width)
+    cropped_images, cropped_masks, grain_crop_coordinates = prep_arrays(image, grains_mask, pad_width)
```

Comment:
If getting the `grain_crop_coordinates` is done using a dedicated function then would also need...

```python
    grain_crop_coordinates = extract_grain_anchor(grains_mask)
```

Reply:
Thanks so much for the feedback on the PR @ns-rse!

Not sure I'm understanding your comment completely, but it seems to me that the crop coordinates/offsets need to be calculated and extracted from the crop_array function because of the padding that crop_array adds being unpredictable due to it increasing the size of the cropped image (and therefore the appropriate grain coordinate/offset) by different amounts depending on if a grain is near the border, so it has to calculate the appropriate offsets to return inside the function.

So I don’t think I could make a separate function for calculating the grain offsets without re-writing the same image-edge proximity logic inside crop_array..?

And so `prep_arrays` needs to handle it because it contains the call to `crop_array`?

Reply:
Here's a comparison of the current method and using the bounding boxes with
```python
[(grain.bbox[0], grain.bbox[1]) for grain in region_properties]
```
current method:
<img width="743" alt="image" src="https://github.com/AFM-SPM/TopoStats/assets/86117496/73616508-44f4-4e0c-8980-d61baf553dac">

bounding boxes:
<img width="759" alt="image" src="https://github.com/AFM-SPM/TopoStats/assets/86117496/8d45fabc-ba5a-4f9f-9a43-f429ec726262">

the bounding boxes traces are offset by the pad_width but only when far from the top left image borders as defined by `crop_array`, so I found I could only get it to behave right if `crop_array` told us by how much it pads each grain (and thus the offset)

It is really messy 😞 but  I wasn't able to think of a better solution myself